src/docs/arch-session.dox

```diff
@@ -0,0 +1,44 @@
+/*! @arch_page arch-session Session
+
+@section arch_session_def Definition
+After a @ref arch-connection has been established between the application and WiredTiger, the
+application can start sending requests to WiredTiger using a session. A session is internally
+represented by WT_SESSION and plays an important role since almost all operations require one.
+
+A session can only be created through an existing connection with the API
+WT_CONNECTION::open_session. It is possible to create multiple sessions through the same connection.
+In fact, one connection can have multiple sessions but one session can only be associated to one
+connection.
+
+@section arch_session_ops Operations
+The different operations a session can request to WiredTiger are related to cursors handles, table
+operations and transactions. You can read the complete description of each possible operation in the
+documentation related to WT_SESSION.
+
+@section arch_session_txn Transactions
+It is possible to group a number of operations within a session and get a single result, in other
+words, multiple operations can be treated as a single atomic operation. This can be done using @ref
+arch-transaction. When multiple operations are part of a transaction, they must either all be
+successful or fail; WiredTiger will cancel any successful operation before a failure within the same
```

Comment:
cancel -> abort? "successful" is confusing.

Reply:
I changed `cancel` to `abort` in 339e6d00b48db9f60bd28f04c4046825929dcb6a.
I left successful as I think it sounds correct. Can we have more thoughts about it ? @bvpvamsikrishna @AlexeyAnisimovAU